app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt

```diff
@@ -50,43 +55,86 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.dp
 import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.address.Address
+import com.github.se.cyrcle.model.address.AddressViewModel
 import com.github.se.cyrcle.model.map.MapViewModel
 import com.github.se.cyrcle.model.parking.Parking
 import com.github.se.cyrcle.model.parking.ParkingViewModel
 import com.github.se.cyrcle.model.user.UserViewModel
+import com.github.se.cyrcle.permission.PermissionHandler
+import com.github.se.cyrcle.ui.map.MapConfig
 import com.github.se.cyrcle.ui.navigation.NavigationActions
 import com.github.se.cyrcle.ui.navigation.Route
 import com.github.se.cyrcle.ui.navigation.Screen
 import com.github.se.cyrcle.ui.theme.atoms.ScoreStars
 import com.github.se.cyrcle.ui.theme.atoms.Text
 import com.github.se.cyrcle.ui.theme.molecules.BottomNavigationBar
 import com.github.se.cyrcle.ui.theme.molecules.FilterPanel
+import com.mapbox.geojson.Point
 import com.mapbox.turf.TurfMeasurement
 import kotlin.math.roundToInt
 
 const val CARD_HEIGHT = 120
 const val MAX_SWIPE_DISTANCE = 150
+const val maxSuggestionDisplayNameLengthList = 70
+const val NumberOfSuggestionsForMenu = 6
 
+@SuppressLint("StateFlowValueCalledInComposition")
 @Composable
 fun SpotListScreen(
     navigationActions: NavigationActions,
     parkingViewModel: ParkingViewModel,
     mapViewModel: MapViewModel,
-    userViewModel: UserViewModel
+    userViewModel: UserViewModel,
+    addressViewModel: AddressViewModel,
+    permissionHandler: PermissionHandler
 ) {
+
   val userPosition by mapViewModel.userPosition.collectAsState()
 
   val filteredParkingSpots by parkingViewModel.closestParkings.collectAsState()
   val pinnedParkings by parkingViewModel.pinnedParkings.collectAsState()
 
-  LaunchedEffect(userPosition) { parkingViewModel.setCircleCenter(userPosition) }
+  // chosen location by the user for the list Screen
+  val chosenLocation: MutableState<Address> = remember {
+    mutableStateOf(
+        Address(
+            latitude = MapConfig.defaultCameraState().center.latitude().toString(),
+            longitude = MapConfig.defaultCameraState().center.longitude().toString()))
+  }
+
+  // value that says wether the user clicked on my Location Suggestion or not
+  val myLocation = remember { mutableStateOf(true) }
+
+  // location permission from location manager
+  val locPermission = permissionHandler.getLocalisationPerm().value
+
+  LaunchedEffect(userPosition, myLocation, chosenLocation.value) {
+
+    // if suggestion MyLocatio is chosen and user has location permission then set the circle center
+    // to user position
+    if (locPermission && myLocation.value) parkingViewModel.setCircleCenter(userPosition)
+
+    // else if the user has chosen a location or hasn't given his location permission then set the
+    // circle center to the chosen location (default position is EPFL)
+    else Log.e("SpotListScreen", parkingViewModel.closestParkings.value.toString())
```

Comment:
This log suggests an error. I think this should be a debug log with `Log.d` unless I haven't understood

Reply:
This is the case eveywhere in the code but i will still do it if it matter for you

Reply:
Really? I never noticed 🤭